csharp/Api/Answer/IConceptMap.cs

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using System.Collections.Generic;
+
+using TypeDB.Driver.Api;
+
+namespace TypeDB.Driver.Api
+{
+    /**
+     * Contains a mapping of variables to concepts.
+     */
+    public interface IConceptMap
+    {
+        /**
+         * Returns a collection of all variables in this <code>IConceptMap</code>.
+         *
+         * <h3>Examples</h3>
+         * <pre>
+         * conceptMap.Variables;
+         * </pre>
+         */
+        IEnumerable<string> Variables { get; }
```

Comment:
@dmitrii-ubskii I was thinking about it, but I wasn't really sure. 

Is `Pinvoke.typedb_driver.concept_map_get_variables(NativeObject)``` a network call as well? If so, it looks like we barely have any property in our project (not telling if it's a bad thing).

Reply:
`concept_map_get_variables` is not a network call, it's a getter. Anything that returns a promise is a network call, and iterators _usually_ are network streams, but not always (like, indeed, `concept_map_get_variables`).
The rule of thumb could be: if it makes sense to cache the response on the C# side, it can be a property.

Reply:
Well, if we decide to rename it (with @alexjpwalker signal as well, I guess), I'd like to collect a list of all such cases and approve this list before making changes with you. Would be a good training as well.

Reply:
Decided and renamed. Updated the original comment.